src/Morphader.cpp

```diff
@@ -0,0 +1,324 @@
+					const float crossfadeCV = clamp(inputs[CV_INPUT + i].getVoltage(), 0.f, 10.f);
+					channelCrossfades[i] = rescale(crossfadeCV, 0.f, 10.f, -1.f, +1.f);
+				}
+				// if channel 1 is plugged in, but this channel isn't, channel 1 is normalled - in
+				// this scenario, however the CV is summed with the crossfader
+				else if (inputs[CV_INPUT + 0].isConnected()) {
+					const float crossfadeCV = clamp(inputs[CV_INPUT + 0].getVoltage(), 0.f, 10.f);
+					channelCrossfades[i] = params[CV_PARAM].getValue() * rescale(crossfadeCV, 0.f, 10.f, 0.f, +2.f) + masterCrossfadeValue;
+				}
+				else {
+					channelCrossfades[i] = masterCrossfadeValue;
+				}
+			}
+
+			channelCrossfades[i] = clamp(channelCrossfades[i], -1.f, +1.f);
+		}
+
+		return channelCrossfades;
+	}
+
+	void process(const ProcessArgs& args) override {
+
+		int maxChannels = 1;
+		simd::float_4 mix[4] = {0.f};
+		const simd::float_4 channelCrossfades = determineChannelCrossfades(args.sampleTime);
+
+		for (int i = 0; i < NUM_MIXER_CHANNELS; i++) {
+
+			const int channels = std::max(std::max(inputs[A_INPUT + i].getChannels(), inputs[B_INPUT + i].getChannels()), 1);
+			// keep track of the max number of channels for the mix output, noting that if channels are taken out of the mix
+			// (i.e. they're connected) they shouldn't contribute to the mix polyphony calculation
+			if (!outputs[OUT + i].isConnected()) {
+				maxChannels = std::max(maxChannels, channels);
+			}
+
+			simd::float_4 out[4] = {0.f};
+			for (int c = 0; c < channels; c += 4) {
+				simd::float_4 inA = inputs[A_INPUT + i].getNormalVoltageSimd(normal10VSimd, c) * params[A_LEVEL + i].getValue();
+				simd::float_4 inB = inputs[B_INPUT + i].getNormalVoltageSimd(normal10VSimd, c) * params[B_LEVEL + i].getValue();
+
+				switch (static_cast<CrossfadeMode>(params[MODE + i].getValue())) {
+					case CV_MODE: {
+						out[c / 4] = equalSumCrossfade(inA, inB, channelCrossfades[i]);
+						break;
+					}
+					case AUDIO_MODE: {
+						// in audio mode, close to the centre point it is possible to get large voltages
+						// (e.g. if A and B are both 10V const). however according to the standard, it is
+						// better not to clip this https://vcvrack.com/manual/VoltageStandards#Output-Saturation
+						out[c / 4] = equalPowerCrossfade(inA, inB, channelCrossfades[i]);
+						break;
+					}
+					default: assert(false);
+				}
+			}
+
+			// if output is patched, the channel is taken out of the mix
+			if (outputs[OUT + i].isConnected() && i != NUM_MIXER_CHANNELS - 1) {
+				outputs[OUT + i].setChannels(channels);
+				for (int c = 0; c < channels; c += 4) {
+					out[c / 4].store(outputs[OUT + i].getVoltages(c));
+				}
+			}
+			else {
+				for (int c = 0; c < channels; c += 4) {
+					mix[c / 4] += out[c / 4];
+				}
+			}
+
+			if (i == NUM_MIXER_CHANNELS - 1) {
+				outputs[OUT + i].setChannels(maxChannels);
+
+				for (int c = 0; c < maxChannels; c += 4) {
+					mix[c / 4].store(outputs[OUT + i].getVoltages(c));
```

Comment:
Use `getVoltageSimd()`

Reply:
I'm not sure I follow - does `getVoltageSimd()` not give you float_4 instance (rather than a pointer)? I want to store mix[c/4] in output OUT + i. I got this snippet from VCA:

https://github.com/VCVRack/Fundamental/blob/v1/src/VCA.cpp#L87

Reply:
Ah, `setVoltageSimd(mix[c/4], c)`